docs/discovery.md

```diff
@@ -42,10 +42,34 @@ class MyClass
 }
 ```
 
+## HTTP Async Client Discovery
+
+This type of discovery finds a HTTP Async Client implementation.
+
+``` php
+use Http\Client\HttpAsyncClient;
+use Http\Discovery\HttpAsyncClientDiscovery;
+
+class MyClass
+{
+    /**
+     * @var HttpAsyncClient
+     */
+    protected $httpAsyncClient;
+
+    /**
+     * @param HttpAsyncClient|null $httpAsyncClient to do HTTP requests.
+     */
+    public function __construct(HttpAsyncClient $httpAsyncClient = null)
+    {
+        $this->httpAsyncClient = $httpAsyncClient ?: HttpAsyncClientDiscovery::find();
+    }
+}
+```
 
 ## PSR-7 Message Factory Discovery
 
-This type of discovery finds installed [PSR-7](http://www.php-fig.org/psr/psr-7/) Message implementations and their [factories](message-factory.md).
+This type of discovery finds a [PSR-7](http://www.php-fig.org/psr/psr-7/) Message implementation and their [factories](message-factory.md).
```

Comment:
If it is "a" implementation, then it can't be "their factories". Singulars and plurals should be synchronized. Same for URI.